src/Block/Adminhtml/Form/Field/CustomStrategies.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+namespace Meanbee\ServiceWorker\Block\Adminhtml\Form\Field;
+
+class CustomStrategies extends \Magento\Config\Block\System\Config\Form\Field\FieldArray\AbstractFieldArray
+{
+    /** @var CachingStrategy */
+    protected $strategyRenderer;
+
+    /**
+     * @inheritdoc
+     */
+    protected function _prepareToRender()
+    {
+        $this->addColumn("path", [
+            "label" => __("URL Path"),
+        ]);
+
+        $this->addColumn("strategy", [
+            "label"    => __("Strategy"),
+            "renderer" => $this->getStrategyRenderer(),
+        ]);
+
+        $this->_addAfter = false;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    protected function _prepareArrayRow(\Magento\Framework\DataObject $row)
+    {
+        $optionName = sprintf("option_%s", $this->getStrategyRenderer()->calcOptionHash($row->getData("strategy")));
+
+        $row->setData("option_extra_attrs", [
+            $optionName => 'selected="selected"',
+        ]);
+    }
+
+    /**
+     * Get the caching strategy input field renderer.
+     *
+     * @return CachingStrategy
+     */
+    protected function getStrategyRenderer()
+    {
+        if (!$this->strategyRenderer) {
+            $this->strategyRenderer = $this->getLayout()->createBlock(
+                "Meanbee\\ServiceWorker\\Block\\Adminhtml\\Form\\Field\\CachingStrategy",
```

Comment:
Don't know if it would work exactly the same way, but I use this form for referencing classes in other projects: `Meanbee\ServiceWorker\Block\Adminhtml\Form\Field\CachingStrategy::class`.